run_southafrica.py

```diff
@@ -1,95 +1,129 @@
+    """Main entry point allowing external calls
+
+    Args:
+      args ([str]): command line parameter list
+    """
+    args = parse_args(args)
+    if args.path is not None:
+        download_path = str(args.path)
     else:
-        new_za.delete_unneeded_files('duplicates-nomatch-'+specs,[],files_path=new_za.downloadPath+'/mandates')
-    new_za.delete_unneeded_files('duplicates-'+specs,[])
-    new_za.teardown()
-elif args.webpage == 3: 
-    matches_acts = new_za.run_acts(keywords)
-    specs = 'SouthAfrica-acts'
-    if len(matches_acts) > 0:
-        print(matches_acts) 
-        new_za.print_matches(matches_acts,specs)
-        new_za.delete_no_matches(specs,path=new_za.downloadPath+'/acts')
+        current_dir = Path(os.path.abspath(os.curdir))
+        download_path = current_dir / 'south_africa_output'
+        download_path = str(download_path)  # class wants strings for paths
+        if not os.path.exists(download_path):
+            os.mkdir(download_path)
+
+    # setting up options for Chromedriver
+    # mostly to ensure any PDF links automatically
+    # download the PDFs to download_path
+    options = webdriver.ChromeOptions()
+    options.add_experimental_option('prefs', {
+        "download.default_directory": download_path,  # Default dir downloads
+        "download.prompt_for_download": False,  # to auto download the file
+        "download.directory_upgrade": True,
+        "plugins.always_open_pdf_externally": True  # don't show PDF in chrome
+    })
+
+    pd.options.display.max_colwidth = 1000
+    keywords = ['legal aid']  # TODO: take these as command-line options?
+
+    # keywords = ['legal aid',
+    #           'legal assistance',
+    #           'legal service',
+    #           'judicial assistance']
+
+    new_za = legisSouthAfrica(args.driver, download_path, options)
+    new_za.checkers()
+
+    if args.webpage == 1:
+        scrape(new_za, keywords, 'constit')
+    elif args.webpage == 2:
+        scrape(new_za, keywords, 'mandates')
+    elif args.webpage == 3:
+        scrape(new_za, keywords, 'acts')
+    elif args.webpage == 4:
+        scrape(new_za, keywords, 'bills')
     else:
-        new_za.delete_unneeded_files('duplicates-nomatch-'+specs,[],files_path=new_za.downloadPath+'/acts')
-    new_za.delete_unneeded_files('duplicates-'+specs,[])
-    new_za.teardown()
-elif args.webpage == 4: 
-    matches_bills = new_za.run_bills(keywords)
-    specs = 'SouthAfrica-bills'
-    if len(matches_bills) > 0:
-        print(matches_bills) 
-        new_za.print_matches(matches_bills,specs)
-        new_za.delete_no_matches(specs,path=new_za.downloadPath+'/bills')
+        error_msg = 'ERROR: webpage integer indicator not found.'
+        raise ValueError(error_msg + webpage_help)
+
+
+def scrape(new_za, keywords: list, page_type: str):
+    matches = new_za.run_bills(keywords)
+    specs = 'SouthAfrica-' + page_type
+    download_path = new_za.downloadPath + '/' + page_type
+    if matches:
```

Comment:
this part was honestly a necessary refactor and I'm glad you did it.

Reply:
I'd like to set up proper path-handling using `pathlib.Path`, which I started to do but gave up on once I realized your functions needed strings. Nice thing about it is that it handles nonsense like trailing slashes, various root directories.

Reply:
I'd be curious to see how this works, but indeed, I liberally employ strings. I try to do some slash handling using `os` methods that know how to check for that. ^^;